core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -405,9 +405,9 @@ private[spark] object HadoopRDD extends Logging {
    * The three methods below are helpers for accessing the local map, a property of the SparkEnv of
    * the local process.
    */
-  def getCachedMetadata(key: String): Any = SparkEnv.get.hadoopJobMetadata.get(key)
+  def getCachedMetadata(key: String): AnyRef = SparkEnv.get.hadoopJobMetadata.get(key)
```

Comment:
Problem here was using a non-ref type in a Java Map. This is only really used for a JobConf anyway